ansible/roles/test/tasks/pfc_wd/functional_test/functional_test.yml

```diff
@@ -0,0 +1,299 @@
+- conn_graph_facts: host={{ peer_device }}
+  connection: local
+
+- name: Prepare variables required for PFC test
+  set_fact:
+    pfc_queue_index: 4
+    pfc_frames_number: 2000000
+    pfc_fanout_interface: "{{neighbors[pfc_wd_test_port]['peerport']}}"
+    peer_hwsku: "{{device_info['HwSku']}}"
+    testname: functional_test
+    config_files:
+      - pfc_wd_drp_action.json
+      - pfc_wd_fwd_action.json
+      - pfc_wd_del_action.json
+
+- set_fact: pfc_wd_test_neighbor_addr="{{ pfc_wd_test_port_addr | regex_replace('(^.*\.).*$', '\\1') }}{{pfc_wd_test_port_addr.split('.')[3] | int + 1 }}"
+- set_fact: pfc_wd_rx_neighbor_addr="{{ pfc_wd_rx_port_addr | regex_replace('(^.*\.).*$', '\\1') }}{{pfc_wd_rx_port_addr.split('.')[3] | int + 1 }}"
+
+- set_fact:
+      peer_login: "{{switch_login[hwsku_map[peer_hwsku]]}}"
+
+- name: Set timers
+  set_fact:
+      pfc_wd_poll_time: 200
+      pfc_wd_penalty: 30000
```

Comment:
rename to restoration time? Set it to 30s looks to large to me. Can we also test the case when setting it to a relative small value eg. 200ms?

Reply:
renames is nice. although i see no issue with the current names. 
for testing normal flow we should use the values provided by Azure production
for testing different flows 9if any) we can use any value.
what the purpose to change the 30000 to 200ms? this is not the case for production.

Reply:
I would prefer to rename as it will be easier for others to read when the variable names exactly match their meaning.  For the timer setting, my suggestion is to test both large value and small value. As when the time setting is in ms, it pose harder challenge to our system than the large value. Just want to make sure it also works when the setting is small. 

BTW, according to the shared requirement doc, the recovery time should be <200ms. Let me know if I have missed something.

Reply:
I will rename them. As to requirements, recovery time here is set to 30 seconds to verify traffic is forwarded/dropped while the queue is in the stormed state. I will make smaller time value for timer test.